### PR TITLE
559-refactor: Widget places

### LINKED ISSUE #559: FSD: Widget Places
## 📝 Describe the refactoring task

Part of the issue #347 

## 📍 Code location

`src/widgets/places`

### places.module.scss
1. Refactor `places.tsx`'s `scss` as following

-   remove all parent selector `& .card-content` -> `.card-content`. 
-   change all possible values to variables if such exists.
-   and everything in your opinion that needs to be refactored
-   replace all element selectors by `tag` with `class` selector `img` -> `.image`

### places.tsx
1. replace `div`'s with `HTML5` tags
2. move `places` const to `constants.ts`
3. remove `string[]` type
4. get rid of `Marquee` library

### places.test.tsx
1. move `test` file to `ui` folder
2. remove redundant `places` constant

### PR DESCRIPTION
**Pull request by Quiddlee**
## What type of PR is this? (select all that apply)

- [ ] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 🚧 Breaking Change
- [x] 🧑‍💻 Code Refactor
- [ ] 📝 Documentation Update

## Related Tickets & Documents

- Related Issue #559 
- Closes #559 

## Screenshots, Recordings
Screenshot 2024-11-25 at 12 04 16 PM

## Added/updated tests?

- [x] 👌 Yes
- [ ] 🙅‍♂️ No, because they aren't needed
- [ ] 🙋‍♂️ No, because I need help

## [optional] What gif best describes this PR or how it makes you feel?
RidingGIF

## Release Notes

- **New Features**
	- Introduced a new `Places` component that displays a list of geographical locations with a marquee effect.
	- Added a `PlaceItem` component for rendering individual place entries.
	- New styling for the `Places` component to enhance layout and responsiveness.
	- Added a `MarqueeGroup` component for improved marquee functionality.

- **Bug Fixes**
	- Removed the deprecated `react-double-marquee` dependency, ensuring compatibility with updated React versions.

- **Tests**
	- Implemented a new test suite for the `Places` component to verify rendering behavior and layout correctness.

- **Chores**
	- Updated project dependencies to use the latest versions of React and Node.js.
	- Introduced a constant for place names to streamline data management.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -29,7 +29,6 @@
     "next": "15.0.2",
     "react": "19.0.0-rc-02c0e824-20241028",
     "react-dom": "19.0.0-rc-02c0e824-20241028",
-    "react-double-marquee": "^1.1.0",
```
Comment (SpaNb4) [inline package.json:32]: nice!

File: src/widgets/places/ui/places/places.module.scss [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    display: flex;
+
+    padding: 5px 20px;
+
+    border-color: $color-gray-600;
+    border-style: solid;
+    border-width: 1px 0 1px 0;
+
+    .marquee-group {
+      will-change: transform;
```
Comment (SpaNb4) [inline src/widgets/places/ui/places/places.module.scss:15]: Do we really need this property?

Comment (Quiddlee) [reply src/widgets/places/ui/places/places.module.scss]: removed

File: src/widgets/places/ui/places/places.tsx [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+      <div aria-hidden="true" className={cx('marquee-group')}>
+        {places.map((place) => (
+          <PlaceItem key={place}>{place}</PlaceItem>
+        ))}
+      </div>
+      <div aria-hidden="true" className={cx('marquee-group')}>
+        {places.map((place) => (
+          <PlaceItem key={place}>{place}</PlaceItem>
+        ))}
+      </div>
```
[the remark this answers is not in the document]
Comment (Quiddlee) [reply src/widgets/places/ui/places/places.tsx]: done

File: src/widgets/places/ui/places/places.module.scss [hunk 4]
```diff
@@ -0,0 +1,24 @@
+.places {
+  white-space: nowrap;
+
+  &.content {
```
Comment (YulikK) [inline src/widgets/places/ui/places/places.module.scss:4]: I thought you asked to remove `& .card-content` -> `.card-content` in all ishiu's

Comment (Quiddlee) [reply src/widgets/places/ui/places/places.module.scss:4]: seems to be resolved https://github.com/rolling-scopes/site/pull/661#discussion_r1876810571

**Review by KristiBo — APPROVED** [review]

**Pull request merged**